src/main/kotlin/no/nb/bikube/core/controller/CoreController.kt

```diff
@@ -59,25 +58,104 @@ class CoreController (
     @Operation(
-        summary = "Search for title by name",
+        summary = "Search catalogue",
         description =
-            "Case insensitive search producing a list of titles matching the search string. " +
+            "Case insensitive search on various types (titles, publishers, items etc.) and various material types. " +
             "Supports wildcard operator * for partial matches."
     )
     @ApiResponses(value = [
         ApiResponse(responseCode = "200", description = "OK"),
         ApiResponse(responseCode = "400", description = "Bad request"),
         ApiResponse(responseCode = "500", description = "Server error")
     ])
-    fun getTitleByName(
-        @RequestParam name: String,
-        @RequestParam materialType: MaterialType,
-    ): ResponseEntity<Flux<Title>> {
-        return when(materialTypeToCatalogueName(materialType)) {
-            CatalogueName.COLLECTIONS -> ResponseEntity.ok(axiellService.getTitleByName(name))
-            else -> throw NotSupportedException("Material type $materialType is not supported.")
+    fun search(
+        @RequestParam searchTerm: String,
+        @RequestParam searchType: SearchType,
+        @RequestParam materialType: MaterialType
+    ): ResponseEntity<Flux<CatalogueRecord>> {
+        if (searchTerm.isEmpty()) throw BadRequestBodyException("Search term cannot be empty.")
+
+        return when (searchType) {
+            SearchType.TITLE -> {
+                when(materialTypeToCatalogueName(materialType)) {
+                    CatalogueName.COLLECTIONS -> ResponseEntity.ok(axiellService.searchTitleByName(searchTerm))
+                    else -> throw NotSupportedException("Material type $materialType is not supported.")
+                }
+            }
+            SearchType.PUBLISHER -> {
+                when(materialTypeToCatalogueName(materialType)) {
+                    CatalogueName.COLLECTIONS -> ResponseEntity.ok(axiellService.searchPublisherByName(searchTerm))
+                    else -> throw NotSupportedException("Material type $materialType is not supported.")
+                }
+            }
+            SearchType.LANGUAGE -> {
+                when(materialTypeToCatalogueName(materialType)) {
+                    CatalogueName.COLLECTIONS -> ResponseEntity.ok(axiellService.searchLanguageByName(searchTerm))
+                    else -> throw NotSupportedException("Material type $materialType is not supported.")
+                }
+            }
+            SearchType.LOCATION -> {
+                when(materialTypeToCatalogueName(materialType)) {
+                    CatalogueName.COLLECTIONS -> ResponseEntity.ok(axiellService.searchPublisherPlaceByName(searchTerm))
+                    else -> throw NotSupportedException("Material type $materialType is not supported.")
+                }
+            }
+            else -> throw NotSupportedException("Search type $searchType is not supported.")
+        }
+    }
```

Comment:
As below, might not need to actually search for language/publisher/place in the API.